net/v2ray/Makefile

```diff
@@ -0,0 +1,297 @@
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_ROUTING),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/app\/router"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_STATISTICS),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/app\/stats"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_BLACKHOLE_PROTO),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/proxy\/blackhole"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_DNS_PROXY),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/proxy\/dns"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_DOKODEMO_PROTO),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/proxy\/dokodemo"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_FREEDOM_PROTO),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/proxy\/freedom"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_MTPROTO_PROXY),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/proxy\/mtproto"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_HTTP_PROTO),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/proxy\/http"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_SHADOWSOCKS_PROTO),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/proxy\/shadowsocks"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_SOCKS_PROTO),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/proxy\/socks"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_VMESS_PROTO),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/proxy\/vmess\/inbound"/\/\/ &/; \
+	s/_ "v2ray.com\/core\/proxy\/vmess\/outbound"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_TCP_TRANS),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/transport\/internet\/tcp"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_MKCP_TRANS),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/transport\/internet\/kcp"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_WEBSOCKET_TRANS),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/transport\/internet\/websocket"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_HTTP2_TRANS),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/transport\/internet\/http"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_DOMAIN_SOCKET_TRANS),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/transport\/internet\/domainsocket"/\/\/ &/;
+endif
+
+ifeq ($(CONFIG_V2RAY_DISABLE_QUIC_TRANS),y)
+V2RAY_SED_ARGS += \
+	s/_ "v2ray.com\/core\/transport\/internet\/quic"/\/\/ &/;
+endif
+
+endif
+
+define Build/Prepare
+	$(Build/Prepare/Default)
+	( \
+		sed -i \
+			's/\(version[[:space:]]*=[[:space:]]*"\).*\("\)/\1$(PKG_VERSION)\2/; \
+			s/\(build[[:space:]]*=[[:space:]]*"\).*\("\)/\1OpenWrt - Release $(PKG_RELEASE)\2/' \
+			$(PKG_BUILD_DIR)/core.go ; \
+	)
+ifneq ($(V2RAY_SED_ARGS),)
+	( \
+		sed -i \
+			'$(V2RAY_SED_ARGS)' \
+			$(PKG_BUILD_DIR)/main/distro/all/all.go ; \
+	)
+endif
+ifneq ($(CONFIG_V2RAY_EXCLUDE_ASSETS),y)
+	wget -O $(PKG_BUILD_DIR)/release/config/geoip.dat \
+		https://github.com/v2ray/geoip/releases/latest/download/geoip.dat
+
+	wget -O $(PKG_BUILD_DIR)/release/config/geosite.dat \
+		https://github.com/v2ray/domain-list-community/releases/latest/download/dlc.dat
```

Comment:
[Calling `Download`](https://openwrt.org/docs/guide-developer/packages#download_override) would be a cleaner way to download extra data.

But more importantly, it would be better to download tagged/versioned versions of these files. Using the latest releases would not be reproducible (they can change between builds).

And actually, would it make sense/be possible to package these files in a separate package, instead of including them conditionally based on a config option?

Reply:
These two files are part of v2ray